File: sites/impact_test.js
const accountSid = "IR9Yfq6ikDt47136577kPXBmtJoZopM5i1";
const authToken = "wcXTBcRmcy5aMPmpp3zi--nkmZyozpiX";

const auth = Buffer.from(`${accountSid}:${authToken}`).toString('base64');
const baseUrl = `https://api.impact.com/Partners/${accountSid}`;

async function impactApi() {
  try {
    console.log(`Testing Account: ${accountSid}`);
    const accountRes = await fetch(baseUrl, {
      headers: {
        'Authorization': `Basic ${auth}`,
        'Accept': 'application/json'
      }
    });
    console.log(`Account Status: ${accountRes.status}`);
    if (accountRes.ok) {
      console.log(JSON.stringify(await accountRes.json(), null, 2));
    } else {
      console.log(await accountRes.text());
    }

    console.log("\nGetting Media Properties...");
    const propertiesRes = await fetch(`${baseUrl}/MediaProperties`, {
      headers: {
        'Authorization': `Basic ${auth}`,
        'Accept': 'application/json'
      }
    });
    console.log(`Media Properties Status: ${propertiesRes.status}`);
    if (propertiesRes.ok) {
      console.log(JSON.stringify(await propertiesRes.json(), null, 2));
    }
  } catch (error) {
    console.error("Error:", error);
  }
}

impactApi();

File: sites/impact_test.py
import requests
from requests.auth import HTTPBasicAuth
import json

account_sid = "IR9Yfq6ikDt47136577kPXBmtJoZopM5i1"
auth_token = "wcXTBcRmcy5aMPmpp3zi--nkmZyozpiX"

base_url = f"https://api.impact.com/Partners/{account_sid}"

if __name__ == "__main__":
    headers = {"Accept": "application/json"}
    
    # Test Account
    print(f"Testing Account: {account_sid}")
    r = requests.get(base_url, auth=HTTPBasicAuth(account_sid, auth_token), headers=headers)
    print(f"Account Status: {r.status_code}")
    if r.status_code == 200:
        print(json.dumps(r.json(), indent=2))
        
    # Get Media Properties
    print("\nGetting Media Properties...")
    r = requests.get(f"{base_url}/MediaProperties", auth=HTTPBasicAuth(account_sid, auth_token), headers=headers)
    print(f"Media Properties Status: {r.status_code}")
    if r.status_code == 200:
        print(json.dumps(r.json(), indent=2))
